### PR TITLE
test/helpers: log to console when report generation begins / ends

### PR DESCRIPTION
**Pull request by ianvernon**
It is extremely difficult to determine at what point in Console output on Jenkins when
a test failed because if it fails, a bunch of information about commands being ran is logged to console.  Instead of logging this information to the file in which logs are stored,
log it to console directly to make it easier for people to see exactly when tests failed,
as this provides key information to narrow down where to look for information in logs,
endpoint state changes, etc.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
Example from running a test locally in which I added an `Expect(false).To(BeTrue())` at the end to invoke failure:

```
Ians-MBP:test ianvernon$ ginkgo --focus="RuntimeValidatedPolicies" --focus="L3-Dependent L7 Egress" -v -noColor -- -cilium.provision=false
level=info msg="environment variable \"K8S_VERSION\" was not set; setting to default value \"1.10\""
Running Suite: runtime
======================
Random Seed: 1524245017
Will run 1 of 278 specs

SSS
------------------------------
RuntimeValidatedPolicies 
  L3-Dependent L7 Egress
  /Users/ianvernon/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:312
time="2018-04-20T10:23:47-07:00" level=info msg=Starting test=RunPolicies
time="2018-04-20T10:23:47-07:00" level=info msg="Vagrant: running command 'vagrant ssh-config runtime'"
time="2018-04-20T10:23:50-07:00" level=debug msg="generated SSHConfig for node runtime"
time="2018-04-20T10:23:50-07:00" level=debug msg="done importing ssh config"
time="2018-04-20T10:23:50-07:00" level=debug msg="running command: ls /tmp/"
time="2018-04-20T10:23:56-07:00" level=debug msg="running command: sudo cilium status"
time="2018-04-20T10:23:56-07:00" level=info msg="Cilium status is true" test=RunPolicies
time="2018-04-20T10:23:56-07:00" level=debug msg="running command: sudo docker network create --ipv6 --subnet ::1/112 --driver cilium --ipam-driver cilium cilium-net"
time="2018-04-20T10:23:57-07:00" level=info msg="setting PolicyEnforcement=default"
time="2018-04-20T10:23:57-07:00" level=debug msg="running command: sudo cilium config -o json | jq -r '.status.realized[\"policy-enforcement\"]'"
time="2018-04-20T10:23:57-07:00" level=debug msg="running command: sudo cilium config PolicyEnforcement=default"
time="2018-04-20T10:23:57-07:00" level=debug msg="spinning up container with command 'docker run -d --name httpd1 --net cilium-net -l id.httpd1 cilium/demo-httpd'"
time="2018-04-20T10:23:57-07:00" level=debug msg="running command: sudo docker run -d --name httpd1 --net cilium-net -l id.httpd1 cilium/demo-httpd"
time="2018-04-20T10:23:58-07:00" level=debug msg="spinning up container with command 'docker run -d --name httpd2 --net cilium-net -l id.httpd2 cilium/demo-httpd'"
time="2018-04-20T10:23:58-07:00" level=debug msg="running command: sudo docker run -d --name httpd2 --net cilium-net -l id.httpd2 cilium/demo-httpd"
time="2018-04-20T10:24:00-07:00" level=debug msg="spinning up container with command 'docker run -d --name httpd3 --net cilium-net -l id.httpd3 cilium/demo-httpd'"
time="2018-04-20T10:24:00-07:00" level=debug msg="running command: sudo docker run -d --name httpd3 --net cilium-net -l id.httpd3 cilium/demo-httpd"
time="2018-04-20T10:24:01-07:00" level=debug msg="spinning up container with command 'docker run -d --name app1 --net cilium-net -l id.app1 tgraf/netperf'"
time="2018-04-20T10:24:01-07:00" level=debug msg="running command: sudo docker run -d --name app1 --net cilium-net -l id.app1 tgraf/netperf"
time="2018-04-20T10:24:04-07:00" level=debug msg="spinning up container with command 'docker run -d --name app2 --net cilium-net -l id.app2 tgraf/netperf'"
time="2018-04-20T10:24:04-07:00" level=debug msg="running command: sudo docker run -d --name app2 --net cilium-net -l id.app2 tgraf/netperf"
time="2018-04-20T10:24:06-07:00" level=debug msg="spinning up container with command 'docker run -d --name app3 --net cilium-net -l id.app3 tgraf/netperf'"
time="2018-04-20T10:24:06-07:00" level=debug msg="running command: sudo docker run -d --name app3 --net cilium-net -l id.app3 tgraf/netperf"
time="2018-04-20T10:24:08-07:00" level=info msg="Deleting all policy in agent"
time="2018-04-20T10:24:08-07:00" level=debug msg="running command: sudo cilium policy delete --all -o json | jq '.revision'"
time="2018-04-20T10:24:08-07:00" level=debug msg="running command: sudo cilium policy wait 9"
time="2018-04-20T10:24:16-07:00" level=debug msg="running command: cilium endpoint list -o jsonpath='{range [*]}{@.status.external-identifiers.container-name}{\"=\"}{@.status.state}{\"\\n\"}{end}'"
time="2018-04-20T10:24:16-07:00" level=info msg="'7' containers are in a 'ready' state of a total of '7' containers." functionName=WaitEndpointsReady status="map[ready:7]" test=RunPolicies
time="2018-04-20T10:24:16-07:00" level=info msg="setting PolicyEnforcement=default"
time="2018-04-20T10:24:16-07:00" level=debug msg="running command: sudo cilium config -o json | jq -r '.status.realized[\"policy-enforcement\"]'"
time="2018-04-20T10:24:17-07:00" level=debug msg="running command: sudo cilium config PolicyEnforcement=default"
time="2018-04-20T10:24:22-07:00" level=debug msg="running command: cilium endpoint list -o jsonpath='{range [*]}{@.status.external-identifiers.container-name}{\"=\"}{@.status.state}{\"\\n\"}{end}'"
time="2018-04-20T10:24:22-07:00" level=info msg="'7' containers are in a 'ready' state of a total of '7' containers." functionName=WaitEndpointsReady status="map[ready:7]" test=RunPolicies
STEP: Setting up policy: /home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json
time="2018-04-20T10:24:22-07:00" level=debug msg="running command: sudo cilium policy get -o json | jq '.revision'"
time="2018-04-20T10:24:22-07:00" level=info msg="before importing policy" file-path=/home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json policyRevision=9 test=RunPolicies
time="2018-04-20T10:24:22-07:00" level=info msg="validating policy before importing" file-path=/home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json test=RunPolicies
time="2018-04-20T10:24:22-07:00" level=debug msg="running command: sudo cilium policy validate /home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json"
time="2018-04-20T10:24:22-07:00" level=debug msg="running command: sudo cilium policy import /home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json"
time="2018-04-20T10:24:28-07:00" level=debug msg="running command: sudo cilium policy get -o json | jq '.revision'"
time="2018-04-20T10:24:29-07:00" level=debug msg="running command: sudo cilium policy wait 10"
time="2018-04-20T10:24:29-07:00" level=debug msg="running command: sudo cilium policy get -o json | jq '.revision'"
time="2018-04-20T10:24:29-07:00" level=info msg="policy import finished and revision increased" file-path=/home/vagrant/go/src/github.com/cilium/cilium/test/runtime/manifests/Policies-l3-dependent-l7-egress.json policyRevision=10 test=RunPolicies
time="2018-04-20T10:24:29-07:00" level=debug msg="running command: sudo cilium endpoint list -o jsonpath='{range [?(@.status.labels.security-relevant[0]!=\"reserved:health\")]}{@.status.external-identifiers.container-name}{\"=\"}{@.id}{\"\\n\"}{end}'"
time="2018-04-20T10:24:30-07:00" level=debug msg="running command: sudo docker inspect app3"
time="2018-04-20T10:24:30-07:00" level=debug msg="running command: sudo docker inspect httpd1"
STEP: Client "app3" attempting to curl public URL on httpd1
time="2018-04-20T10:24:30-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://10.15.167.158:80/public"
STEP: Client "app3" attempting to curl public URL on httpd1
time="2018-04-20T10:24:30-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::a0f:0:0:ecfe]:80/public"
time="2018-04-20T10:24:30-07:00" level=debug msg="running command: sudo cilium endpoint get 10687 -o json"
time="2018-04-20T10:24:31-07:00" level=debug msg="running command: sudo docker inspect app3"
time="2018-04-20T10:24:31-07:00" level=debug msg="running command: sudo docker inspect httpd1"
STEP: Client "app3" attempting to curl private URL on httpd1
time="2018-04-20T10:24:31-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://10.15.167.158:80/private"
STEP: Client "app3" attempting to curl private URL on httpd1
time="2018-04-20T10:24:31-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::a0f:0:0:ecfe]:80/private"
time="2018-04-20T10:24:31-07:00" level=debug msg="running command: sudo cilium endpoint get 10687 -o json"
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: sudo docker inspect app3"
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: sudo docker inspect httpd2"
STEP: Client "app3" attempting to curl public URL on httpd2
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://10.15.195.213:80/public"
STEP: Client "app3" attempting to curl public URL on httpd2
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::a0f:0:0:3b14]:80/public"
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: sudo cilium endpoint get 10687 -o json"
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: sudo docker inspect app3"
time="2018-04-20T10:24:32-07:00" level=debug msg="running command: sudo docker inspect httpd2"
STEP: Client "app3" attempting to curl private URL on httpd2
time="2018-04-20T10:24:33-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://10.15.195.213:80/private"
STEP: Client "app3" attempting to curl private URL on httpd2
time="2018-04-20T10:24:33-07:00" level=debug msg="running command: docker exec -i  app3 curl -s --fail --connect-timeout 5 --max-time 5 http://[f00d::a0f:0:0:3b14]:80/private"
time="2018-04-20T10:24:33-07:00" level=debug msg="running command: sudo cilium endpoint get 10687 -o json"
time="2018-04-20T10:24:33-07:00" level=debug msg="running command: sudo journalctl -au cilium --since '46.273878539 seconds ago'"
time="2018-04-20T10:24:33-07:00" level=error msg="failed to kill command: EOF"
===================== TEST FAILED =====================
time="2018-04-20T10:24:33-07:00" level=debug msg="running command: sudo cilium endpoint list"
time="2018-04-20T10:24:34-07:00" level=debug msg="running command: cilium kvstore get cilium --recursive"
time="2018-04-20T10:24:34-07:00" level=debug msg="running command: cilium service list -o json"
time="2018-04-20T10:24:34-07:00" level=debug msg="running command: cilium config"
time="2018-04-20T10:24:35-07:00" level=debug msg="running command: sudo cilium bpf lb list"
time="2018-04-20T10:24:35-07:00" level=debug msg="running command: sudo cilium bpf ct list global"
time="2018-04-20T10:24:35-07:00" level=debug msg="running command: sudo cilium bpf tunnel list"
time="2018-04-20T10:24:35-07:00" level=debug msg="running command: cilium policy get"
time="2018-04-20T10:24:35-07:00" level=debug msg="running command: cilium status --all-controllers"
time="2018-04-20T10:24:36-07:00" level=debug msg="running command: cilium endpoint list -o json"
time="2018-04-20T10:24:36-07:00" level=debug msg="running command: sudo cilium-bugtool -t /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/RuntimeValidatedPoliciesL3-DependentL7Egress"
time="2018-04-20T10:24:49-07:00" level=debug msg="running command: sudo journalctl -au cilium --no-pager"
time="2018-04-20T10:24:49-07:00" level=debug msg="running command: sudo journalctl -au cilium-docker --no-pager"
time="2018-04-20T10:24:49-07:00" level=debug msg="running command: sudo docker logs cilium-consul"
time="2018-04-20T10:24:49-07:00" level=debug msg="running command: sudo rsync -rv --exclude=*.sock /var/run/cilium /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/RuntimeValidatedPoliciesL3-DependentL7Egress/lib"
time="2018-04-20T10:24:49-07:00" level=debug msg="running command: sudo rsync -rv --exclude=*.sock /var/lib/cilium /home/vagrant/go/src/github.com/cilium/cilium/test/test_results/RuntimeValidatedPoliciesL3-DependentL7Egress/run"
time="2018-04-20T10:24:50-07:00" level=debug msg="running command: docker ps -aq"
time="2018-04-20T10:24:50-07:00" level=debug msg="running command: docker logs 6c1e803f25e7"
time="2018-04-20T10:24:50-07:00" level=debug msg="running command: docker logs 61dcb402c6fc"
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs 18d19d503c69"
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs acd97b336c00"
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs "
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs c9e6431275c2"
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs f20f0dc404a2"
time="2018-04-20T10:24:51-07:00" level=debug msg="running command: docker logs 6deda81200d6"
time="2018-04-20T10:24:52-07:00" level=debug msg="running command: docker logs a990b70d77a2"
time="2018-04-20T10:24:52-07:00" level=debug msg="running command: docker logs 12ce994348ac"
time="2018-04-20T10:24:52-07:00" level=debug msg="running command: sudo journalctl -au cilium | grep -qi -B 5 -A 5 deadlock"
===================== EXITING REPORT GENERATION =====================
time="2018-04-20T10:24:52-07:00" level=info msg="Deleting all policy in agent"
time="2018-04-20T10:24:52-07:00" level=debug msg="running command: sudo cilium policy delete --all -o json | jq '.revision'"
time="2018-04-20T10:24:52-07:00" level=debug msg="running command: sudo cilium policy wait 11"
time="2018-04-20T10:24:53-07:00" level=info msg="Deleting all policy in agent"
time="2018-04-20T10:24:53-07:00" level=debug msg="running command: sudo cilium policy delete --all -o json | jq '.revision'"
time="2018-04-20T10:24:53-07:00" level=debug msg="running command: sudo cilium policy wait 11"
time="2018-04-20T10:24:56-07:00" level=debug msg="removing container with command 'docker rm -f httpd1'"
time="2018-04-20T10:24:56-07:00" level=debug msg="running command: sudo docker rm -f httpd1"
time="2018-04-20T10:24:56-07:00" level=debug msg="removing container with command 'docker rm -f httpd2'"
time="2018-04-20T10:24:56-07:00" level=debug msg="running command: sudo docker rm -f httpd2"
time="2018-04-20T10:24:57-07:00" level=debug msg="removing container with command 'docker rm -f httpd3'"
time="2018-04-20T10:24:57-07:00" level=debug msg="running command: sudo docker rm -f httpd3"
time="2018-04-20T10:24:57-07:00" level=debug msg="removing container with command 'docker rm -f app1'"
time="2018-04-20T10:24:57-07:00" level=debug msg="running command: sudo docker rm -f app1"
time="2018-04-20T10:24:58-07:00" level=debug msg="removing container with command 'docker rm -f app2'"
time="2018-04-20T10:24:58-07:00" level=debug msg="running command: sudo docker rm -f app2"
time="2018-04-20T10:24:58-07:00" level=debug msg="removing container with command 'docker rm -f app3'"
time="2018-04-20T10:24:58-07:00" level=debug msg="running command: sudo docker rm -f app3"
time="2018-04-20T10:24:59-07:00" level=info msg="build is not running on Jenkins; environment variable 'JENKINS_HOME' is not set"
time="2018-04-20T10:24:59-07:00" level=info msg="build is not running on Jenkins; environment variable 'NODE_NAME' is not set"
time="2018-04-20T10:24:59-07:00" level=info msg="AfterSuite: not running on Jenkins; leaving VMs running for debugging"

• Failure [71.658 seconds]
RuntimeValidatedPolicies
/Users/ianvernon/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:280
  L3-Dependent L7 Egress [It]
  /Users/ianvernon/go/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:312

  Expected
      <bool>: false
  to be true

  /Users/ianvernon/go/src/github.com/cilium/cilium/test/runtime/Policies.go:719
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
```

It is now  extremely easy to tell where the test failed in the output and where the logs are gathered.

**Comment by ianvernon** [comment]
test-me-please

File: test/helpers/cilium.go [hunk 1]
```diff
@@ -574,7 +574,7 @@ func (s *SSHMeta) PolicyWait(revisionNum int) *CmdRes {
 // purposes.
 func (s *SSHMeta) ReportFailed(commands ...string) {
 	wr := s.logger.Logger.Out
-	fmt.Fprint(wr, "===================== TEST FAILED =====================\n")
+	fmt.Println("===================== TEST FAILED =====================")
```
Comment (aanm) [inline test/helpers/cilium.go]: This won't be printed in the logger, can you make a `fmt.Fprintf` to print it in the logs?

**Review by aanm — DISMISSED** [review]

File: test/helpers/cilium.go (hunk 1 above)
Comment (ianvernon) [reply test/helpers/cilium.go]: OK, I'll keep the old one.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
IMO, this is ready to merge because despite it failing tests, this is no way related to the policy tests :( . https://github.com/cilium/cilium/issues/3845 has been filed to track the failure of the L3-dependent L7 egress test.

**Comment by ianvernon** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Pull request merged**